src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -432,50 +462,100 @@ private Slave prepareDockerPlatforms() throws Exception {
+        Controller c = script.launch(envVars, ws, launcher, listener);
+        awaitCompletion(c);
+        assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
+        Long timeCheck1 = binaryPath.lastModified();
+
+        c = script.launch(envVars, ws, launcher, listener);
+        awaitCompletion(c);
+        assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
+        Long timeCheck2 = binaryPath.lastModified();
+        assertEquals(timeCheck1, timeCheck2);
+
+        binaryPath.delete();
+        binaryPath.touch(Instant.now().toEpochMilli());
+        try {
+            c = script.launch(envVars, ws, launcher, listener);
+        } catch (Exception e) {
+            assertThat(e, instanceOf(IOException.class));
+            assertThat(e.getMessage(), containsString("Cannot run program"));
+        }
+    }
+
+    private static String readPluginVersion() throws IOException {
+        String pluginVersion = null;
+        try (FileReader pomReader = new FileReader("pom.xml")) {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            Model model = reader.read(pomReader);
+            pluginVersion = model.getProperties().getProperty("revision");
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        }
+        return pluginVersion;
     }
 
     /**
-     * Returns the first zombie process it finds. This method can only be used on tests that launch a script that executes
-     * a `sleep` call and where the end of the `sleep` function also indicates the end of the script.
-     * Additionally, this function cannot be run on the NATIVE platform as the testing conditions are too generic and may be
-     * triggered by some other, unrelated, process.
+     * Returns the first zombie process it finds.
      *
      * @return String `ps` line output of the zombie process found. Empty string otherwise.
      * @throws InterruptedException
      * @throws IOException
      */
-    private String getZombies() throws InterruptedException, IOException {
-        switch(platform) {
-            // debian slim image does not contain PS
+    private void assertNoZombies() throws InterruptedException, IOException {
+        String exitString = null;
+        String zombieString = null;
+        switch (platform) {
             case SLIM:
-            // (See JENKINS-58656) Running in a container with no init process is guaranteed to leave a zombie. Just let this test pass.
+                // Debian slim does not have ps
             case NO_INIT:
-            // Run only on docker platforms as NATIVE `ps` output cannot guarantee it will detect processes correctly. Just let this test pass.
-            case NATIVE:
-                return "";
+                // (See JENKINS-58656) Running in a container with no init process is guaranteed to leave a zombie.
+                // let test pass, otherwise it will mark entire test as skipped
+                assumeTrue(true);
```

Comment:
Looks like a debug line or something?
```suggestion
```

Reply:
actually, that's not a debug line.  What I wanted to have the tests skipped for `SLIM` and `NO_INIT`, but the problem is that `assertZombies` is used in other tests, like `smoke`. If i just skip this test with something like `assumeTrue(!platform.equals(NO_INIT))` then the entire `smoke` test will show as skipped.

Reply:
What I mean though is that `assumeTrue(true)` does nothing, 100% of the time, so `assumeTrue(true); return;` is completely equivalent to `return;`, right?

Reply:
Yeah, you're completely correct. I don't know what my brain was thinking  :p

Reply:
Done in bea4658.